### ISSUE TITLE
Add support for inline --ignore-subject expressions

### ISSUE THREAD

**Issue by dgollahon** [mbj/mutant#999]
While #998 would be a clear improvement over only CLI flag support, I would strongly prefer having the ability to apply inline mutation disables. This helps future readers of the code know to operate with care when they edit a region of code that is not mutation tested and it additionally makes it much easier to keep ignore subjects in sync (for instance, if a subject is renamed or removed) when the disable is inline.

ex:

```ruby
class Foo
  # mutant:ignore
  def bar
    # Don't mutate this
  end
end
```

Supporting method-based disabled would be a great start, but I think that supporting a class/module level or a file level would be very helpful as well as individual lines. I have sometimes had code where it's painful to kill one minor mutation (ex: `#first` -> `#last` on a 0 or 1 element array) and it has been nice to pull that into its own line and disable that one expression (with `mutest`) but keep the rest of the method covered.

The specific syntax and granularity have pros/cons so I am not expecting full agreement, but some kind of inline disable syntax would make `mutant` a much more ergonomic tool and better support incremental adoption.

**Comment by mbj** [comment]
I agree to this feature.

But I'd amend it with the following:

```
# mutant:skip - do not execute at all, equivalent of your feature request
# mutant:uncovered - execute but expect subject to be uncovered / fail if suddenly covered
```
I like the `skip` terminology more as `ignore` could also be read as "ignore uncovered mutations". Skip implies to actually skip over the subject.

**Comment by dgollahon** [comment]
@mbj I completely agree with you. I almost referenced that idea, but saw that #971 already covered the principle and thought it would be simpler to scope the issue to feature parity with the CLI interface until #971 is implemented.

**Comment by mbj** [comment]
@dgollahon Yeah, I'm fine to track them separately at this point.

**Comment by mbj** [comment]
This was just merged and released as v0.11.8. See: #1320
